File: Yogscart Alpha/Assets/Scripts/Track Editor/DeathCatch.js
#pragma strict

var DeathParticles : ParticleSystem;

function OnTriggerEnter (other : Collider) {
if(other.name == "DeathCatch"){

DeathParticles.Play();

if(transform.GetComponent(Position_Finding) != null)
{
if(transform.GetComponent(kartScript) != null)
transform.GetComponent(kartScript).ExpectedSpeed = 0;

yield WaitForSeconds(0.75);

GetComponent.<Rigidbody>().isKinematic = true;

if(transform.GetComponent(kartScript) != null){
var td = GameObject.Find("Track Manager").transform.GetComponent(TrackData);

var nPos = td.PositionPoints[transform.GetComponent(Position_Finding).currentPos].position;
if(transform.GetComponent(Position_Finding).currentPos+1 < td.PositionPoints.Length)
var n1Pos = td.PositionPoints[transform.GetComponent(Position_Finding).currentPos+1].position;
else
n1Pos = td.PositionPoints[0].position;

transform.position = nPos;

transform.rotation = Quaternion.Euler(0,0,0);
transform.rotation = Quaternion.LookRotation(n1Pos-nPos,Vector3.up);
}

GetComponent.<Rigidbody>().isKinematic = false;
}
yield WaitForSeconds(0.1);


DeathParticles.Stop();

}
}

File: Yogscart C#/Assets/Scripts/Kart Scripts/DeathCatch.cs
using UnityEngine;
using System.Collections;

public class DeathCatch : MonoBehaviour {

    public ParticleSystem deathParticles;

	void OnTriggerEnter(Collider other)
    {
	    if(other.name == "DeathCatch")
        {
            deathParticles.Play();

            if(transform.GetComponent<PositionFinding>() != null && transform.GetComponent<KartMovement>() != null)
                transform.GetComponent<KartMovement>().expectedSpeed = 0;

            StartCoroutine("DoRespawn");

        }
	}

    IEnumerator DoRespawn()
    {
        yield return new WaitForSeconds(0.75f);

        GetComponent<Rigidbody>().isKinematic = true;

        if(transform.GetComponent<KartMovement>() != null)
        {
            PositionFinding pf = transform.GetComponent<PositionFinding>();

            Vector3 newPos = pf.closestPoint.transform.position;

            PointHandler nextPoint = pf.closestPoint.connections[0];
            for(int i = 1; i < pf.closestPoint.connections.Count; i++)
            {
                PointHandler ph = pf.closestPoint.connections[i];
                if (ph.percent > nextPoint.percent)
                    nextPoint = ph;
            }

            transform.position = newPos;
            transform.rotation = Quaternion.LookRotation(nextPoint.transform.position - newPos, Vector3.up);

            transform.GetComponent<KartMovement>().tricking = false;
        }

        GetComponent<Rigidbody>().isKinematic = false;
        yield return new WaitForSeconds(0.1f);

        deathParticles.Stop();
    }

}
